packages/ag-charts-community/src/chart/toolbar/toolbar.ts

```diff
@@ -285,16 +302,16 @@ export class Toolbar extends BaseModuleInstance implements ModuleInstance {
     }
 
     private onFloatingAnchorChanged(event: ToolbarFloatingAnchorChangedEvent) {
-        const {
-            elements,
-            groupButtons,
-            positions,
-            horizontalSpacing,
-            verticalSpacing,
-            ctx: { domManager, toolbarManager },
-        } = this;
+        const { elements, positions, horizontalSpacing, verticalSpacing } = this;
+
+        const { group, anchor, index } = event;
+
+        if (this.index == index && this.dragState.detached) {
+            return;
+        }
 
-        const { group, anchor } = event;
+        this.index = index;
+        this.dragState.detached = false;
```

Comment:
This `index` is very specific to annotations and a bit of a random number to save here in toolbar. It's a bit leaky to have it here.

I _think_ you could achieve the same behaviour of reattaching the floating toolbar when you select a new annotation by resetting `detached` in the `Toolbar.toggleVisibilities()` function, something like:

```diff
for (const position of TOOLBAR_POSITIONS) {
    const visible = this.enabled && Array.from(this.positions[position].values()).some(isGroupVisible);
    this.elements[position].classList.toggle(styles.modifiers.hidden, !visible);
+   if (position === ToolbarPosition.Floating && !visible) {
+       this.dragState.detached = false;
+   }
}
```

As when you select a new annotation it will deselect the previous one first, hide the overlays, toggling the visibility of the floating toolbar to false and reattaching it here, then show the new version of it.

Reply:
Discussed offline and decided to fix in refactor of toobar later